### ISSUE TITLE
Random faction/settings/map

### ISSUE THREAD

**Issue by kgersen** [kgersen/Allegiance#169]
Add a button (probably available only to the GC) to randomize the faction, settings and map.

Randomize faction should be straight forward, but settings and map could do with an option to limit the pool it chooses from. Like "don't take into account maps with more than x sectors" etc. Same with settings so the absurd money options like "outrageous" is not on the random pool.

All of this should be set to "random" and the players will only know what the game selected until the game started.

 _Imported from trac ticket [#170](https://trac.alleg.net//ticket/170),  created by anonymous on 07-23-2010 at 03:30, last modified: 07-21-2012 at 17:33_

**Comment by kgersen** [comment]
yea, every game seems to have a random option, it would suit this one as well.  You don't say where you'd like this buttan.

_Trac comment by Imago on 07-23-2010 at 04:33_

**Comment by kgersen** [comment]
Added a couple of examples on this topic:
[http://www.freeallegiance.org/forums/index.php?showtopic=58257]

_Trac comment by anonymous on 07-24-2010 at 01:31_

**Comment by kgersen** [comment]
Awesome!  This gives me an idea of exactly what is needed.

_Trac comment by Imago on 07-24-2010 at 02:13_

**Comment by kgersen** [comment]
I looked at this once, but didn't get anywhere.  Imma try again.

_Trac comment by Xynth on 08-13-2010 at 21:04_

**Comment by kgersen** [comment]
r971

My patch has a bug.  It momentarily shows a random faction's watermark, but it goes away and it isn't the final random faction.  I'm guessing this has to do with the first faction getting there before the random bool gets in the IsideIGC object.  The faction is re-randomized every 5 seconds.

_Trac comment by Xynth on 08-14-2010 at 02:16_

**Comment by kgersen** [comment]
Probably worth linking this here: r982

_Trac comment by TurkeyXIII on 11-11-2010 at 10:15_

**Comment by kgersen** [comment]
I know this is being delayed to R7 now, but might as well mention these bugs I guess:

In addition to showing the wrong faction's logo on the countdown screen at the start of a game, this feature will also show the same wrong logo when hitting F6 once the game has begun. So, for example, you might see the Rix logo and start message when the game is in count-down at launch, and once the game starts you will see the Rix logo for your own team when you hit F6, but you may then realise that actually, you are Giga, with Giga ships and bases and tech, etc.

Also, your pods will be those of the faction on the start screen and in F6, rather than the faction you actually are. So, in the above example, if you get podded as Giga, you will find yourself floating home in a Rixian Unity pod. Not sure if this is just the wrong visual, or if the pod actually behaves as that of the wrong faction.

_Trac comment by girlyboy parry.lost@gmail.com on 11-11-2010 at 14:56_

**Comment by kgersen** [comment]
2 things:
1, the faction was re-randomising even after the countdown started. Some elements of the faction choice are loaded at the start of the countdown, so that was messing with it.

2, after a message was recieved about a civ change, OnTeamCivChange was being triggered before the client set the random civ bool, so the bitmaps (among other things, probably) weren't being updated properly.

_Trac comment by TurkeyXIII on 05-30-2011 at 17:27_

**Comment by kgersen** [comment]
Just briefly tested this. It came up with Dreg vs Omicron. Dreg had Omicron bases and ships (insignia seemed correct). Omicron had BIOS ships. Still not working looks like.

_Trac comment by Spunky on 02-07-2012 at 06:26_

**Comment by kgersen** [comment]
3rd thing:  Server was getting its side ID from the person sending the message, not the one in the message. So it'd only change the GCs faction, and always to team 2's random selection.

_Trac comment by TurkeyXIII on 02-08-2012 at 17:20_

**Comment by kgersen** [comment]
YES! It works! Awesome job!
pkk, can we get this in?

_Trac comment by Spunky on 02-15-2012 at 04:06_

**Comment by kgersen** [comment]
In [1072/allegsvn]:

```
#CommitTicketReference repository="allegsvn" revision="1072"
Finally fixes #170.
```

_Trac comment by TurkeyXIII on 07-21-2012 at 17:33_
